Cookbook/Proposals/AdditionalRulesForDanger.md

```diff
@@ -0,0 +1,133 @@
+
+Here's a brief example of how it was typically done in both Builders and FlowControllers:
+
+```swift
+viewController
+            .reactive
+            .signal(for: #selector(UIViewController.viewWillDisappear))
+            .discardValues()
+            .observeValues(dismiss)
+```
+
+Nowadays, this can be done much more elegantly (thanks Anders!):
+
+```swift
+func send(_ event: ScreenLifecycleEvent) {
+        switch (event, state.status) {
+        case (.didAppear, .showing):
+            runSomething()
+        default:
+            break
+        }
+    }
+```
+
+As of June 7th, 2019, we still have 16 results in 13 different files.
+
+In order to detect this, we could simply monitor occurences of the following regex:
+
+`\.signal\(for: \#selector\(UIViewController\.`
+
+We'd have to restrict this rule to Builders and FlowControllers only, therefore we'd have to whitelist filenames that matched `.*Builder.swift` and `.*FlowController.swift`.
+
+This rule could be enforced with either tool but I'd recommend using SwiftLint for this since due to the low number of warnings it would trigger in the current state of our database. Also, I believe we should try to use Danger for particularly important warnings in order not to spam the pull request with comments.
+
+
+### Deprecated calls of DesignLibrary components (aka pre-NVL components)
+
+Given that NVL is scheduled for release within two weeks, using NVL components is pivotal to Babylon's overall look and feel.
+With the exception of the empty space component (even though it has its own NVL equivalent), every other pre-NVL component used inside Renderer should trigger a warning.
+
+The following regex currently yields 1333 results in 216 files: `Component\.(?!EmptySpace)\b`.
+The sheer number of warnings immediately excludes SwiftLint; Danger would be our only option.
+
+Nevertheless, this would require that everyone relied exclusively on NVL components and I am not sure this is the case right now.
+
+
+### Mutating Current in `setUp()` but not restoring it in `tearDown()`
+
+The rule is simple: every object (susceptible to global mutation) that is mutated in `setUp()` must be restored in `tearDown()`.
+Mutations in `Current` in particular are the prime suspects of a number of flaky tests that have been recently affecting our test suite.
+
+In order to detect this problem we have to use Danger since SwiftLint only accepts regular expressions.
+We would have to whitelist every test file (`.*Tests.*`) and use Danger to manually parse the test line-by-line until the `setUp()` function was found. 
+
+If found, we need to make a counter that increases by one each time a curly bracket and does the reverse when a closing curly bracket appears. Once the counter reaches zero again, then we've successfully delimited the contents of the `setUp()` function and we'd use the following regex to detect changes in Current: `Current\..*\ = `.
```

Comment:
We need to also detect `Current.changeTheme` or other `Current.setXXX`, not just `Current = `

Reply:
Actually, @daniloaliberti pointed out that this issue is [already covered](https://github.com/Babylonpartners/babylon-ios/pull/7806/files#diff-74895f4da31ec40d83d342bf612540b4R17) 👍

I will remove this section entirely

Reply:
> Can I suggest to add the rule that would validate the PR metadata itself, its title at least 🙏 Currently we have a bot that will fail if there is no properly formatted ticket number in the title, but it does not explain the failure. We can extend this further to validating that if there are no test changes in the PR then there should be a note in the PR description about that

I think that's a good idea @ilyapuchka but we should perhaps make another proposal because this one already has a lot of stuff for just a 15m meeting